torch/utils/data/datapipes/utils/common.py

```diff
@@ -78,18 +78,21 @@ def onerror(err : OSError):
                 dirs.sort()
 
 
-def get_file_binaries_from_pathnames(pathnames: Iterable, mode: str):
+def get_file_binaries_from_pathnames(pathnames: Iterable, mode: str, encoding: str = 'utf-8'):
     if not isinstance(pathnames, Iterable):
         pathnames = [pathnames, ]
 
     if mode in ('b', 't'):
         mode = 'r' + mode
 
+    if 'b' in mode and encoding is not None:
+        raise ValueError("binary mode doesn't take an encoding argument")
+
```

Comment:
Let's move this part into `__init__` function of `FileOpener` to raise Error at the construction time.

Reply:
Do we envision that other classes may use this function eventually? If so, it may be better to do here so we don't have to repeat the checks.

Reply:
No, I don't think there will be. And, even if other classes use such function, they should raise Error at the construction time as well to let users know the Error as early as possible.

Reply:
Yes, good point. I'll make this change.